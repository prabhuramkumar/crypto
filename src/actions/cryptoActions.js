export const UPDATE_CURRENCIES = 'UPDATE_CURRENCIES';

export const updateCurrencies= (val)=>{
    return{
        type: UPDATE_CURRENCIES,
        val
    }
}


