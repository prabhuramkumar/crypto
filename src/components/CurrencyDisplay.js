import React, {useEffect } from 'react';
import cryptoData from '../data/cryptoData';
import currencyAnalysis from '../logics/currencyAnalysis';
import CurrencyDisplayBox from './CurrencyDisplayBox';

const CurrencyDisplay = (props)=>{
    const {currencies, appError} = props.state;
    
	useEffect(() => {
        const result = currencyAnalysis(cryptoData);

        if(result && result.length > 0){
        	props.actions.updateCurrencies(result);
        }else {
        	props.actions.updateError(result);
        }
    }, [props.actions]);

    const generateDisplayBox = () => {
    	if(!appError && currencies.length > 0) {
            return currencies.map((currency, index)=>{
                return(
                    <CurrencyDisplayBox  currency={currency}  key={currency.id}/> 
                )
            })
        }else
            return (<p className="error__message">Some error accoured.</p>)
        }

    return(
        <div className="currencyDisplay">
          <h2 className="currencyDisplay__title">Crypto - Profitable Currencies</h2>
          <div className="currencyDisplay__boxes">
      	     {generateDisplayBox()}
          </div>
        </div>
    )
}

export default CurrencyDisplay;