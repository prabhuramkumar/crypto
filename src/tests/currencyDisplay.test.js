import React from 'react';
import { shallow, mount } from 'enzyme';
import CurrencyDisplay from '../components/CurrencyDisplay';
import * as CryptoActions from '../actions/cryptoActions';

describe('currencyDisplay snapshot', () => {
	const initialState = {
	    currencies:[{
    	  "id": 1,
	      "currency": "BTC",
	      "buyValue": "34.98",
	      "sellValue": "37.01",
	      "buyTime":  "0915",
	      "sellTime": "1230",
	      "date": "20180507",
	      "profit": "2.03"
	    }],
	    appError: ''
	}
  it('should render correctly in normal mode', () => {
    const currencyDisplayBox = shallow(<CurrencyDisplay actions={CryptoActions} state={initialState} />);
  	expect(currencyDisplayBox).toMatchSnapshot();
  });
});

describe('currencyDisplay tests', () => {
  it('should render error when no data', () => {
    const currencies= [];
	const state = {
	    currencies,
	    appError: ''
	}
    const currencyDisplay = mount(<CurrencyDisplay actions={CryptoActions} state={state} />);
  	const component = currencyDisplay.find('.currencyDisplay');
  	const dislayBox = currencyDisplay.find('.currencyDisplay__box');
  	const error = currencyDisplay.find('.error__message');
    expect(component.length).toEqual(1);
    expect(error.length).toEqual(1);
    expect(dislayBox.length).toEqual(0);
  });

  it('should render component when there is data.', () => {
  	const currencies= [{
	  "id": 1,
      "currency": "BTC",
      "buyValue": "34.98",
      "sellValue": "37.01",
      "buyTime":  "0915",
      "sellTime": "1230",
      "date": "20180507",
      "profit": "2.03"
    }]
    const state = {
	    currencies,
	    appError: ''
	}
    const currencyDisplay = mount(<CurrencyDisplay actions={CryptoActions} state={state} />);
  	const component = currencyDisplay.find('.currencyDisplay');
  	const dislayBox = currencyDisplay.find('.currencyDisplay__box');
  	const error = currencyDisplay.find('.error__message');
    expect(component.length).toEqual(1);
    expect(dislayBox.length).toEqual(1);
    expect(error.length).toEqual(0);
  });

   it('should render more component when there is more data.', () => {
  	const currencies= [{
	  "id": 1,
      "currency": "BTC",
      "buyValue": "34.98",
      "sellValue": "37.01",
      "buyTime":  "0915",
      "sellTime": "1230",
      "date": "20180507",
      "profit": "2.03"
    },
    {
	  "id": 2,
      "currency": "BTC",
      "buyValue": "14.98",
      "sellValue": "17.01",
      "buyTime":  "0915",
      "sellTime": "1230",
      "date": "20180507",
      "profit": "2.03"
    }];
    const state = {
	    currencies,
	    appError: ''
	}
    const currencyDisplay = mount(<CurrencyDisplay actions={CryptoActions} state={state} />);
  	const dislayBox = currencyDisplay.find('.currencyDisplay__box');
    expect(dislayBox.length).toEqual(2);
  });
});