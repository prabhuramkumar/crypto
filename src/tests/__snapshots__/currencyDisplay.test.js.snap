// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`currencyDisplay snapshot should render correctly in normal mode 1`] = `
<div
  className="currencyDisplay"
>
  <h2
    className="currencyDisplay__title"
  >
    Crypto - Profitable Currencies
  </h2>
  <div
    className="currencyDisplay__boxes"
  >
    <CurrencyDisplayBox
      currency={
        Object {
          "buyTime": "0915",
          "buyValue": "34.98",
          "currency": "BTC",
          "date": "20180507",
          "id": 1,
          "profit": "2.03",
          "sellTime": "1230",
          "sellValue": "37.01",
        }
      }
      key="1"
    />
  </div>
</div>
`;
